src/fixes.rs

```diff
@@ -0,0 +1,147 @@
+use crate::common::*;
+
+mod lowercase_key;
+
+trait Fix {
+    fn name(&self) -> &str;
+
+    fn fix_warnings(
+        &self,
+        warnings: Vec<&mut Warning>,
+        lines: &mut Vec<LineEntry>,
+    ) -> Option<usize> {
+        let mut count: usize = 0;
+        for warning in warnings {
+            let line = &mut lines[warning.line_number() - 1];
+            if self.fix_line(line).is_some() {
+                warning.set_fixed(true);
+                count += 1;
+            }
+        }
+
+        Some(count)
+    }
+
+    fn fix_line(&self, _line: &mut LineEntry) -> Option<()> {
+        None
+    }
+
+    fn set_fixed_on_all(&self, warnings: Vec<&mut Warning>) -> Option<usize> {
+        let count = warnings.len();
+        for warning in warnings {
+            warning.set_fixed(true);
+        }
+
+        Some(count)
+    }
+}
+
+// TODO: skip fixes (like checks)
+// The fix order is matter
+fn fixlist() -> Vec<Box<dyn Fix>> {
+    vec![
+        // At first we run the fixers that handle a single line entry (they use default
+        // implementation of the fix_warnings() function)
+        Box::new(lowercase_key::LowercaseKeyFixer::default()),
+        // Then we should run the fixers that handle the line entry collection at whole.
+        // And at the end we should run the fixer for ExtraBlankLine check (because the previous
+        // fixers can create additional extra blank lines).
+    ]
+}
+
+pub fn run(warnings: &mut [Warning], lines: &mut Vec<LineEntry>) -> usize {
+    if warnings.is_empty() {
+        return 0;
+    }
+
+    for warning in warnings.iter_mut() {
+        warning.set_fixed(false);
```

Comment:
I think it's enough to make the default field `false` instead of `Option` 
It seems like unspoken logic

Reply:
`None` is used when we are not fixing Warnings, in this case they are no Fixed, no Unfixed.

Reply:
> `None` is used when we are not fixing Warnings, in this case they are no Fixed, no Unfixed.

Just anyway, we see them (`None`) as false here :thinking:

Reply:
true - `Fixed` prefix (we tried and fixed)
false - `Unfixed` prefix (we tried and couldn't fix)
None - no prefix (we didn't try)

Reply:
But as I say, I will take a look on alternative approaches.

Reply:
I changed the output format and there is no need in Option<bool> anymore.